aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -207,6 +224,46 @@ module aptos_framework::fungible_asset {
         object::object_from_constructor_ref<Metadata>(constructor_ref)
     }
 
+    /// Create a fungible asset store whose transfer rule would be overloaded by the provided function.
+    public(friend) fun register_dispatch_functions(
+        constructor_ref: &ConstructorRef,
+        withdraw_function: FunctionInfo,
+		deposit_function: FunctionInfo,
+    ) {
+        let dispatcher_withdraw_function_info = function_info::new_function_info(
+	        @aptos_framework,
+            string::utf8(b"dispatchable_fungible_asset"),
+            string::utf8(b"dispatchable_withdraw"),
+        );
+        // Verify that caller type matches callee type so wrongly typed function cannot be registered.
+        assert!(function_info::check_dispatch_type_compatibility(
```

Comment:
do we ensure the function is public? otherwise it can be upgraded and even change the function signature

Reply:
do we want to be even stricter, and require it to be un-upgradeable?